src/codegen/intrinsic-applications.lisp

```diff
@@ -58,6 +59,19 @@ Returns a new `ast:node'."))
       (t
        child))))
 
+(defmethod transform-intrinsic-application ((rator-name (eql 'coalton:debug-type)) node)
```

Comment:
This is probably too late for the current implementation of intrinsic applications, but we really want the `qualified-type` from the type checker AST in `coalton-impl/typechecker/expression`.